pkgs/tools/backup/znapzend/default.nix

```diff
@@ -0,0 +1,71 @@
+{ stdenv, fetchFromGitHub, zfs, mbuffer, perl, perlPackages, wget, autoconf, automake }:
+
+let
+  version = "v0.15.3";
+  checksum = "1xk0lgb23kv1cl0wc2rav75hjrjigd0cp3hjw9gxab835vsvnkq0";
+in
+stdenv.mkDerivation rec {
+  name = "znapzend-${version}";
+
+  src = fetchFromGitHub{
+    owner = "oetiker";
+    repo = "znapzend";
+    rev = version;
+    sha256 = checksum;
+  };
+
+  buildInputs = [ autoconf automake perl perlPackages.TestHarness perlPackages.Mojolicious
+                  perlPackages.TAPParserSourceHandlerpgTAP perlPackages.MojoIOLoopForkCall
+                  perlPackages.IOPipely
+                  wget ];
+
+  preConfigure = ''
+    sed -i 's/^SUBDIRS =.*$/SUBDIRS = lib/' Makefile.am
+
+    grep -v thirdparty/Makefile configure.ac > configure.ac.tmp
+    mv configure.ac.tmp configure.ac
+
+    autoconf
+  '';
+
+  preBuild = ''
+    aclocal
+    automake
+  '';
+
+  postInstall = ''
+    substituteInPlace $out/bin/znapzend --replace "${perl}/bin/perl" \
```

Comment:
I wonder if it would be easier to generate a wrapper script that sets PERL5LIB or whatever the perl search path variable is?